src/Sylius/Bundle/LocaleBundle/Context/RequestHeaderBasedLocaleContext.php

```diff
@@ -33,17 +38,21 @@ public function getLocaleCode(): string
             throw new LocaleNotFoundException('No main request available.');
         }
 
-        $localeCode = $request->headers->get('Accept-Language');
-        if (null === $localeCode) {
-            throw new LocaleNotFoundException('No locale is set on the master request\'s headers.');
-        }
-
         $availableLocalesCodes = $this->localeProvider->getAvailableLocalesCodes();
-        if (!in_array($localeCode, $availableLocalesCodes, true)) {
-            throw LocaleNotFoundException::notAvailable($localeCode, $availableLocalesCodes);
+
+        // Request::getPreferredLanguage() returns first available locale code if none matches. To allow detection of
+        // this unwanted behavior, we will prepend special locale code to the list of available locale codes.
+        $prependedAvailableLocalesCodes = array_merge(['FIRSTLOCALECODE'], $availableLocalesCodes);
```

Comment:
What would you say for something more meaningful?
```suggestion
        $prependedAvailableLocalesCodes = array_merge(['ALL_CODES_ARE_INVALID_STUB'], $availableLocalesCodes);
```

+ can you put this value into private consts?